.gitignore

```diff
@@ -0,0 +1,12 @@
+/**/__pycache__/
+/**/output.sas
+/**/sas_plan*
+/builds/
+/experiments/issue[0-9]*/.venv/
+/experiments/issue[0-9]*/data/
+/experiments/issue[0-9]*/*-microbenchmark/.obj/
+/experiments/issue[0-9]*/*-microbenchmark/benchmark
+/experiments/issue[0-9]*/*-microbenchmark/Makefile.depend
```

Comment:
Note that these don't have the same semantics as in our previous ignore file: we used to have zero or more digits. Now we have a digit followed by any characters. That's why I said sas_plan.[1-9]* above rather than sas_plan.[0-9]* because these numbers never begin with a 0, and [0-9]* to me looks like it wants to be a regex, which it isn't. Looking at the gitignore doc, I think there is no way to express "zero or more digits" or anything of the sort in the supported syntax. Given that, I think it's cleaner/more idiomatic not to use a half-baked solution that cares about the first digit but not the others and instead use the not-at-all-baked "issue*"  and "sas_plan.*" instead of the versions with the first-digit range.

Not very important as such, but we also want to establish conventions.

Reply:
Note also that it might be nicer to not have these "microbenchmark" rules here and instead have specific .gitignore files only in the relevant issue directories. In hg we didn't have that option. I'm not advocating it necessarily, just mentioning the option. 

I assume the "data" and ".venv" ones perhaps make sense globally, although personally I wouldn't actually want to ignore these files: I want to know if I have "data" directories in my repositories because that's a hint I want to clean them up, not least considering their massive size. Basically, "git status" will tell me which data directories I still have to deal with, I don't want them to hang around in my clone forever. (It's a common impulse to ignore everything that shouldn't be committed, but it's not a dichotomy.) Just a thought, feel free to keep ignoring data and .venv.

Reply:
I see a large advantage in ignoring `data`: After running an experiment, they will contain a large amount of files and if they are not ignored, git operations (like `git status`) will be much slower and have their output cluttered. I understand that you see this as a reason and reminder to not keep the data around there, but so far, I usually did multiple iterations of experiments during which I wanted to keep the experiment data and still work with the repository.

Reply:
I changed the patterns as you suggest. I agree with Florian about `data`. Maybe it's best not to change too many things now. We can always change them when we have more experience with Git.

Reply:
Makes sense regarding `data`.